contracts/src/lib.rs

```diff
@@ -61,6 +61,20 @@ mod tests {
         }
     }
 
+    impl BatchTransport for ChainIdTransport {
```

Comment:
Ideally we can remove all these test transports in favour of mock transport. I'll look into this on Friday.